src/common/adapters/notion.test.js

```diff
@@ -0,0 +1,111 @@
+import client from '../client';
+import loc from './__helpers__/location';
+import scan from './notion';
+
+jest.mock('../client', () => jest.fn());
+
+describe('notion adapter', () => {
+  const id = '5b1d7dd7-9107-4890-b2ec-83175b8eda83';
+  const title = 'Add notion.so support';
+  const slugId = '5b1d7dd791074890b2ec83175b8eda83';
+  const slug = `Add-notion-so-support-${slugId}`;
+  const response = {
+    results: [
+      {
+        role: 'editor',
+        value: {
+          id,
+          type: 'page',
+          properties: { title: [[title]] },
+        },
+      },
+    ],
+  };
+
+  const ticket = { id, title, type: 'page' };
+  const api = { post: jest.fn() };
+
+  beforeEach(() => {
+    api.post.mockReturnValue({ json: () => response });
+    client.mockReturnValue(api);
+  });
+
+  afterEach(() => {
+    api.post.mockReset();
+    client.mockReset();
+  });
+
+  it('returns an empty array when not on a www.notion.so page', async () => {
+    const result = await scan(loc('another-domain.com'));
+    expect(api.post).not.toHaveBeenCalled();
+    expect(result).toEqual([]);
+  });
+
+  it('uses the notion.so api', async () => {
+    const location = loc('www.notion.so', `/notionuser/${slug}`);
+    await scan(location);
+    expect(client).toHaveBeenCalledWith('https://www.notion.so');
+    expect(api.post).toHaveBeenCalled();
+  });
+
+  it('returns an empty array when the current page is a board view', async () => {
+    const res = {
+      results: [
+        {
+          role: 'editor',
+          value: { id, type: 'collection_view_page' },
+        },
+      ],
+    };
+    api.post.mockReturnValueOnce({ json: () => res });
+
+    const location = loc(
+      'www.notion.so',
+      `/notionuser/${slugId}`,
+      '?v=77ff97cab6ff4beab7fa6e27f992dd5e'
+    );
+    const result = await scan(location);
+    const request = { requests: [{ table: 'block', id }] };
+    expect(api.post).toHaveBeenCalledWith('api/v3/getRecordValues', {
+      json: request,
+    });
+    expect(result).toEqual([]);
+  });
+
+  it('returns an emtpy array when the page does not exist', async () => {
+    const res = { results: [{ role: 'editor' }] };
+    api.post.mockReturnValueOnce({ json: () => res });
```

Comment:
Yes, at least for this scenario. There may be cases of "actual" 404s on the API but I did not discover them yet.

I tried out a few different error scenarios with the type of request we're doing by replaying modified requests with `fetch` in the console. Our payload roughly looks like this:
```js
{
  requests: [
    {
      table: 'block',
      id: 'some-uuid'
    }
  ]
}
```
Regardless of what I do to the `id` field value, changing it to an invalid UUID _(not confirm to RFC 4122)_, changing it to a dash or whitespace, I'll always get a `200` success response. An empty one though.

Changing the value of the `table` field, e.g. to `admins` will return a `400` response though. The payload could look somewhat like this:
```js

{
  requests: [
    { table: 'admins', id: 'some-uuid' }
  ]
}
```
The response is then an object with an `errorId`, `message`, etc. I initially also had this as a test scenario, but then realized `ky` is treating status codes outside of the 2xx range as an error anyways 🤷. So I guess it already does everything we need?

Reply:
I see. Thanks a lot for the detailed explanation. 🙇 I was just wondering whether the test should use `mockRejectedValue` exactly because `ky` throws for non-`2xx` responses. But if we get a success, it's perfect as it is. 💚